tesseract_rviz/include/tesseract_rviz/render_tools/trajectory_monitor_widget.h

```diff
@@ -111,6 +111,7 @@ private Q_SLOTS:
 
 public:
   void incomingDisplayTrajectory(const tesseract_msgs::Trajectory::ConstPtr& msg);
+  tesseract_environment::Environment::Ptr env_;
```

Comment:
What about adding a `setEnvironment` or `replaceEnvironment`? This way is is supported here for future use case.

``` c++
void setEnvironment(tesseract_environment::Environment::Ptr env)
{
  onDisable();
  visualization_->initialize(true, true, true, true);
  env_ = std::move(env);
  visualization_->setVisible(true);
  onEnable();
}
```

Reply:
Yeah that sounds good. It strikes me that we might want to separate the displaying code from the subscriber code. Currently there is still a box to select the trajectory topic, but we will probably want to use this same approach to display the task info archive. I'm thinking we make a class that contains all of the code to display the trajectory (looping, trail, etc). Then the PlanningResponseArchive, TaskInfo, and Trajectory displays all contain one and have their own methods of setting the environment, trajectory, etc.

Reply:
I agree.